db/db_impl.cc

```diff
@@ -679,12 +689,45 @@ void DBImpl::PersistStats() {
     stats_history_size_limit = mutable_db_options_.stats_history_buffer_size;
   }
 
-  // TODO(Zhongyi): also persist immutable_db_options_.statistics
-  {
-    std::map<std::string, uint64_t> stats_map;
-    if (!statistics->getTickerMap(&stats_map)) {
-      return;
+  std::map<std::string, uint64_t> stats_map;
+  if (!statistics->getTickerMap(&stats_map)) {
+    return;
+  }
+
+  if (immutable_db_options_.persist_stats_to_disk) {
+    char* timestamp = new char[kNowMicrosStringLength + 1];
```

Comment:
Why not `char timestamp[kNowMicrosStringLength + 1];`?

Reply:
visual studio compiler doesn't like it, see https://ci.appveyor.com/project/Facebook/rocksdb/builds/24222890 for commit https://github.com/facebook/rocksdb/pull/5046/commits/a00f70e89f9af4107d38615ee95e4832a21473af

Reply:
Search the code base, you'll see that it is used in a lot of places. For example:
https://github.com/facebook/rocksdb/blob/05d9d82181547703e5610f423be7da6bfda3ac00/cache/sharded_cache.cc#L134-L139

Reply:
I think it's because MSVS got confused and thought `kNowMicrosStringLength` is not a constant. It might have to do with it being an extern and defined elsewhere

Reply:
Why not make it a const?

Reply:
it is a const, the problem is MSVS thinks otherwise